libsolidity/ast/Types.cpp

```diff
@@ -740,6 +740,18 @@ TypeResult IntegerType::binaryOperatorResult(Token _operator, Type const* _other
 		return this;
 	}
 
+	if (auto const* other = dynamic_cast<RationalNumberType const*>(_other); other)
+	{
+		if (_operator == langutil::Token::Mod || _operator == langutil::Token::Div)
```

Comment:
I would make this the outermost if, since `other` is not used in this condition but in the one below.